### ISSUE TITLE
Input to the propagation network

### ISSUE THREAD

**Issue by zanglam** [seoungwugoh/ivs-demo#2]
Hi, in the paper, you say that: "The two object masks are represented with probabilities and the neutral mask is used if the mask is not available."(Section 3.1 Propagation Network).

However, in the code you write:
```python
def Prop_forward(self, target, end):
        for n in range(target+1, end+1):  #[1,2,...,N-1]
            print('[MODEL: propagation network] >>>>>>>>> {} to {}'.format(n-1, n))
            self.all_E[:,n], _, self.next_a_ref = self.model_P(self.ref, self.a_ref, self.all_F[:,:,n], self.prev_E[:,n], torch.round(self.all_E[:,n-1]), self.dummy_M, [1,0,0,0,0])
```
It seems that you take the probability map(self.prev_E[:,n]) obtained at the previous round and the binary mask(torch.round(self.all_E[:,n-1])) obtained at the previous frame as input to the propagation network. I tried to reproduce the results presented in the paper using your code and the pre-trained model. I would like to know what input should I use.

**Comment by seoungwugoh** [comment]
It was an action only for this demo.
In a real user application, we found that binarized mask is advantageous for robustly handling random user inputs.
In all experiments using DAVIS evaluation server, we used soft-mask as described in the paper.

To reproduce our quantitative results, soft mask without binarization should be used. 
And, you also have to use other pre-trained weights trained using the soft mask (which is not uploaded here).  
If you are interested, I can send you weights, but I'm not sure if it is compatible with this demo code. 
Please, contact me by E-mail.

**Comment by zanglam** [comment]
Thanks for your reply!
